### PR TITLE
plan: selection to dual directly if condition is always false.

### LINKED ISSUE #4977: execute sql with  impossible where

### PR DESCRIPTION
**Pull request by winoros**
This fix #4977 
PTAL @hanfei1991 @lamxTyler @zz-jason

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -409,6 +409,11 @@ func (b *planBuilder) buildSelection(p LogicalPlan, where ast.ExprNode, AggMappe
 				ret, err := expression.EvalBool(expression.CNFExprs{con}, nil, b.ctx)
 				if ret || err != nil {
 					continue
+				} else {
```
Comment (hanfei1991) [inline plan/logical_plan_builder.go]: else if err == nil.

if err != nil , maybe there is a runtime error. Let it report in executor.

**Comment by hanfei1991** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**